lib/tasks/evm.rake

```diff
@@ -55,6 +55,19 @@ namespace :evm do
     EvmApplication.update_stop
   end
 
+  desc "Determine the deployment scenario"
+  task :deployment_status => :environment do
+    status_to_code = {
+      "new_deployment" => 0,
+      "new_replica"    => 1,
+      "redeployment"   => 2,
+      "upgrade"        => 3
```

Comment:
I think we should pick better numbers, but overall I like the approach: http://tldp.org/LDP/abs/html/exitcodes.html